packages/core/src/node/request/proxy.ts

```diff
@@ -0,0 +1,61 @@
+/********************************************************************************
+ * Copyright (C) 2022 TypeFox and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { parse as parseUrl, Url } from 'url';
+import * as httpAgent from 'http-proxy-agent';
+import * as httpsAgent from 'https-proxy-agent';
+
+export type ProxyAgent = httpAgent.HttpProxyAgent | httpsAgent.HttpsProxyAgent;
+
+function getSystemProxyURI(requestURL: Url, env: typeof process.env): string | undefined {
+    if (requestURL.protocol === 'http:') {
+        return env.HTTP_PROXY || env.http_proxy;
+    } else if (requestURL.protocol === 'https:') {
+        return env.HTTPS_PROXY || env.https_proxy || env.HTTP_PROXY || env.http_proxy;
+    }
+
+    return undefined;
+}
+
+export interface ProxySettings {
+    proxyUrl?: string;
+    strictSSL?: boolean;
+}
+
+export function getProxyAgent(rawRequestURL: string, env: typeof process.env, options: ProxySettings = {}): ProxyAgent | undefined {
```

Comment:
It looks like this function is an exact copy of the one in `dev-packages` - can we import that one, instead?

Reply:
Actually, that was a leftover of when I refactored the request-service into its own package. I removed the file.